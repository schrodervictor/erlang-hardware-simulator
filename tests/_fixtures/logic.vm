push constant 0
not
push constant 8
push constant 9
and
push constant 8
push constant 2
or
