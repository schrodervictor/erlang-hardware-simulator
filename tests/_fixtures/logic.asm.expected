// push constant 0
@0
D=A
@SP
A=M
M=D
@SP
M=M+1
// not
@SP
A=M-1
M=!M
// push constant 8
@8
D=A
@SP
A=M
M=D
@SP
M=M+1
// push constant 9
@9
D=A
@SP
A=M
M=D
@SP
M=M+1
// and
@SP
AM=M-1
D=M
A=A-1
M=D&M
// push constant 8
@8
D=A
@SP
A=M
M=D
@SP
M=M+1
// push constant 2
@2
D=A
@SP
A=M
M=D
@SP
M=M+1
// or
@SP
AM=M-1
D=M
A=A-1
M=D|M
